### PR TITLE
code expantion

### PR DESCRIPTION
**Pull request by Einscrew**
Deixei os retornos e os tipos nos argumentos.
Removi os generators. Está minimamente aceitável para o ws.

### CODE REVIEW

File: hangman.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env python3
 
-import random
+from random import choice as random_choise
```
Comment (josecoelho96) [inline hangman.py]: typo in choise : choice

File: hangman.py [hunk 2]
```diff
@@ -68,18 +68,21 @@ def get_random_word() -> str:
...
+            if not words:                
+                print('No words were found in the file specified')
+                exit(-1)
     except FileNotFoundError:
         print('File with words not found!')
         exit(-1)
 
     words = [x.strip() for x in words]
-    return random.choice(words).upper()
+    return random_choise(words).upper()
```
Comment (josecoelho96) [inline hangman.py]: Typo

File: words.txt [hunk 3]
```diff
@@ -1,4 +1,6 @@
 Boa tarde
 Neecist
 Python
-Boa tarde
\ No newline at end of file
+Boa tarde
```
Comment (migueldemoura) [inline words.txt]: This is not compatible with the current codebase. Either remove it or take some of the changes at: https://github.com/migueldemoura/hangman

File: words.txt [hunk 4]
```diff
@@ -1,4 +1,6 @@
 Boa tarde
```
Comment (migueldemoura) [inline words.txt]: This is not compatible with the current codebase. Either remove it or take some of the changes at: https://github.com/migueldemoura/hangman

Comment (migueldemoura) [inline words.txt]: This is not compatible with the current codebase. Either remove it or take some of the changes at: https://github.com/migueldemoura/hangman

File: hangman.py [hunk 5]
```diff
@@ -77,8 +77,10 @@ def get_random_word() -> str:
         print('File with words not found!')
         exit(-1)
 
-    words = [x.strip() for x in words]
-    return random_choice(words).upper()
+    w=[]
+    for x in words[:]:
```
Comment (migueldemoura) [inline hangman.py]: Why the [:]?

File: hangman.py [hunk 6]
```diff
@@ -1,187 +1,150 @@
 #!/usr/bin/env python3
 
-import os
-import random
+from random import choice as random_choice
 import string
 
+draw_size = 35 # odd int
```
Comment (migueldemoura) [inline hangman.py]: Don't use global variables. If this is a global constant, its name should be: DRAW_SIZE

File: hangman.py [hunk 7]
```diff
@@ -1,187 +1,150 @@
...
-    for l in letters[13:]:
+    print_text(firstset)
+
+    for l in initial_letters[set_size//2:]:
         if l in av_letters:
-            s_2 += l + " "
+            secondset += l + ' '
         else:
-            s_2 += "  "
+            secondset += ' '*2
```
Comment (migueldemoura) [inline hangman.py]: Incorrect spacing

File: hangman.py [hunk 8]
```diff
@@ -1,187 +1,150 @@
...
-
-    s_1 = ''
-    for l in letters[:13]:
+    for l in initial_letters[:set_size//2]:
         if l in av_letters:
-            s_1 += l + " "
+            firstset += l + ' '
         else:
-            s_1 += "  "
+            firstset += ' '*2
```
Comment (migueldemoura) [inline hangman.py]: Incorrect spacing. Good place for an inline if, aka poor man's "ternary".

File: hangman.py (hunk 5 above)
Comment (Einscrew) [reply hangman.py]: Era porque antes estava a alterar o próprio words. Mas ya, é inútil. Para efeitos do WS acho que devia constar.

File: hangman.py [hunk 9]
```diff
@@ -1,187 +1,150 @@
...
 
-    printLine()
 
+def get_random_word() -> str:
 
-def getRandomWord():
     try:
         with open('words.txt', 'r') as f:
             words = f.readlines()
+            print(words)
```
Comment (migueldemoura) [inline hangman.py]: Why the print?

File: hangman.py [hunk 10]
```diff
@@ -1,187 +1,150 @@
...
+            if not words:                
+                print('No words were found in the file specified')
+                exit(-1)
```
Comment (migueldemoura) [inline hangman.py]: Error codes shouldn't be the same in both situations.

File: hangman.py [hunk 11]
```diff
@@ -1,187 +1,150 @@
...
-
-
-def printGuesses(word, guesses):
-
-    s = ''
-    for x in word:
-        if x in guesses:
-            s += x + ' '
+    w=[]
+    for x in words[:]:
```
Comment (migueldemoura) [inline hangman.py]: Why the [:]?

File: hangman.py [hunk 12]
```diff
@@ -1,187 +1,150 @@
...
-        printText('Your guess...', False, False)
-        printGuesses(wordToGuess, guesses)
-        printText('Available letters', False, False)
-        printAvailableLetters(available_letters)
-        letter = getLetter(available_letters)
+    failed_attempts = 0
+    has_won = False
+
+    while failed_attempts <= 7:
+        print('\033[H\033[J')
```
Comment (migueldemoura) [inline hangman.py]: Comment needed here. Not obvious for beginners.

File: hangman.py [hunk 13]
```diff
@@ -1,187 +1,150 @@
...
-    printText('hangman', True, True)
-    printHangman(wrongTries)
-    printText('Your guess...', False, False)
-    printGuesses(wordToGuess, guesses)
-    printText('Available letters', False, False)
-    printAvailableLetters(available_letters)
+        cmpWord = list(set(word_to_guess.replace(' ', '')))
+        cmpWord.sort()
+        guesses.sort()
+        if cmpWord == guesses:
```
Comment (migueldemoura) [inline hangman.py]: Don't use an if here... has_won = cmpWord == guesses is simple enough.

File: words.txt [hunk 14]
```diff
@@ -1,5 +1,6 @@
-Ola
-Joaquim
-NEECIST
+Boa tarde
```
Comment (migueldemoura) [inline words.txt]: Spaces aren't supported in this version. Either remove these 2 lines, or backport some changes from: https://github.com/migueldemoura/hangman

File: hangman.py (hunk 10 above)
Comment (Einscrew) [reply hangman.py]: What do you suggest? -2?

File: hangman.py (hunk 9 above)
Comment (Einscrew) [reply hangman.py]: debug

File: hangman.py (hunk 8 above)
Comment (Einscrew) [reply hangman.py]: Não vamos cobrir inlines

File: hangman.py (hunk 11 above)
Comment (Einscrew) [reply hangman.py]: Apenas didático

File: hangman.py (hunk 12 above)
Comment (Einscrew) [reply hangman.py]: Not sure if it even works on windows machines

File: hangman.py (hunk 13 above)
Comment (Einscrew) [reply hangman.py]: Nice one

File: words.txt (hunk 14 above)
Comment (Einscrew) [reply words.txt]: Are you sure?
image

File: hangman.py (hunk 11 above)
Comment (migueldemoura) [reply hangman.py]: There's a .copy() somewhere, remove it and use the [:] instead.

File: hangman.py [hunk 15]
```diff
@@ -1,11 +1,11 @@
...
-
+from os import system
 from random import choice as random_choice
 import string
 
-draw_size = 35 # odd int
-limbs = ['O', ' | ', '/| ', '/|\\', '|', '/  ', '/ \\']
-bodyCombinations = {
+DRAW_SIZE = 35 # odd int
+LIMBS = ['O', ' | ', '/| ', '/|\\', '|', '/  ', '/ \\']
```
Comment (migueldemoura) [inline hangman.py]: LIMBS and BODY_COMBINATIONS should be inside the function that uses them

File: hangman.py [hunk 16]
```diff
@@ -117,7 +116,7 @@ def main() -> None:
     has_won = False
 
     while failed_attempts <= 7:
-        print('\033[H\033[J')
+        system('clear') #cls in windows
```
Comment (migueldemoura) [inline hangman.py]: The previous sequence was alright, no need for a call, especially a os-specific one

File: words.txt (hunk 14 above)
Comment (migueldemoura) [reply words.txt]: That's odd. Can you win the game?

Edit: Nevermind, you also copied the `word_to_guess.replace(' ', '')` for the winning check condition.

File: hangman.py [hunk 17]
```diff
@@ -1,187 +1,149 @@
...
-    printGuesses(wordToGuess, guesses)
-    printText('Available letters', False, False)
-    printAvailableLetters(available_letters)
-
-    if victory:
-        print("Congratz!")
-    else:
-        print("Better luck next time...")
+        cmpWord = list(set(word_to_guess.replace(' ', '')))
+        cmpWord.sort()
```
Comment (migueldemoura) [inline hangman.py]: No need for these 2 sorts. Just use 2 sets.

File: hangman.py [hunk 18]
```diff
@@ -1,187 +1,149 @@
...
-    printText('Your guess...', False, False)
-    printGuesses(wordToGuess, guesses)
-    printText('Available letters', False, False)
-    printAvailableLetters(available_letters)
-
-    if victory:
-        print("Congratz!")
-    else:
-        print("Better luck next time...")
+        cmpWord = list(set(word_to_guess.replace(' ', '')))
```
Comment (migueldemoura) [inline hangman.py]: Remove the list()

**Comment by migueldemoura** [comment]
Might wanna include one of the generators from https://github.com/migueldemoura/hangman. Is there enough workshop time to introduce the word validation at https://github.com/migueldemoura/hangman/blob/master/hangman.py#L50-L51?

File: hangman.py (hunk 18 above)
Comment (Einscrew) [reply hangman.py]: Can we compare _list_ to a _set_?

File: hangman.py (hunk 17 above)
Comment (Einscrew) [reply hangman.py]: Is just for them to use previous learned class methods

File: hangman.py (hunk 18 above)
Comment (Einscrew) [reply hangman.py]: Is just to introduce to some of convertion function there is

File: hangman.py (hunk 11 above)
Comment (Einscrew) [reply hangman.py]: Didn't understand

File: words.txt (hunk 14 above)
Comment (Einscrew) [reply words.txt]: Roger

**Comment by Einscrew** [comment]
I think that the code part will have to be taught in a hurry. And they will have to complete some partes of the code.  But I'll at least one generator. And we probably can put a link to your repository for them to check how can the code be improved

File: hangman.py (hunk 16 above)
Comment (Einscrew) [reply hangman.py]: Luis told me _print('\033[H\033[J')_ on Windows and didn't worked

File: hangman.py (hunk 17 above)
Comment (migueldemoura) [reply hangman.py]: Sure, but stress that it should be a set(a) == set(b) comparison

File: hangman.py (hunk 18 above)
Comment (migueldemoura) [reply hangman.py]: "Sure, but stress that it should be a set(a) == set(b) comparison"

**Comment by migueldemoura** [comment]
Copy the License.md from my repo to here and I'll merge

**Pull request merged**